### PR TITLE
added support for coupons to be used without caps

### PR DESCRIPTION
**Pull request by shreyas-satish**

### CODE REVIEW

File: alembic/versions/2f6a3bb460b8_add_unlimited_to_coupon.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+from alembic import op
+import sqlalchemy as sa
+from sqlalchemy.sql import table, column
+
+
+def upgrade():
+    discount_coupon = table('discount_coupon',
+      column('unlimited', sa.Boolean()))
+    op.add_column('discount_coupon', sa.Column('unlimited', sa.Boolean(), nullable=True))
+    op.execute(discount_coupon.update().values({'unlimited': False}))
```
Comment (jace) [inline alembic/versions/2f6a3bb460b8_add_unlimited_to_coupon.py]: These two steps may be faster with `nullable=False, server_default=False` and then alter to remove `server_default`.

File: boxoffice/models/discount_policy.py [hunk 2]
```diff
@@ -89,7 +90,7 @@ def get_valid_coupons(cls, discount_policies, codes):
         Returns valid coupons, given a list of discount policies and discount codes
         """
         return cls.query.filter(cls.code.in_(codes),
-            cls.used != True,
+            db.or_(cls.used != True, cls.unlimited == True),
```
Comment (jace) [inline boxoffice/models/discount_policy.py]: This needs `# NOQA`

File: boxoffice/models/line_item.py [hunk 3]
```diff
@@ -125,7 +127,7 @@ def apply_discount(self, policy_coupon, line_items, combo=False):
         discount_policy, coupon = policy_coupon
         for line_item in line_items:
             discounted_amount = self.calculate_discounted_amount(discount_policy, line_item)
-            if ((coupon and not coupon_used and discounted_amount > 0) or discount_policy.is_automatic) and (
+            if ((coupon and (coupon.unlimited or not coupon_used)) or discount_policy.is_automatic) and discounted_amount > 0 and (
```
Comment (jace) [inline boxoffice/models/line_item.py]: I can't parse this, but trust your tests cover it.

**Comment by shreyas-satish** [comment]
Closing because using `usage_limit` is a better choice than using `unlimited`. https://github.com/hasgeek/boxoffice/pull/22

**Pull request closed without merging**